From 6f4d7c717a22233195da550e69fa1dfed461f4a9 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 15 Sep 2010 10:42:50 +0100
Patch-mainline: 2.6.37
References: fate#310916
Git-commit: 2f551c84563df2bf144a819993b2d729c66583ee
Subject: [PATCH 1418/2588] drm/i915/sdvo: Mark the status as unknown if
 attached with EDID

One problem with devices that share the DDC bus between the VGA and
DVI-I connectors is that with two devices attached we cannot know if
there is truly a monitor attached to the DVI connector. In this case, it
is preferrrable to mark the status as unknown, so that the user can
supply the known set of modes and continue to use the output.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit 2f551c84563df2bf144a819993b2d729c66583ee)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_sdvo.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_sdvo.c b/drivers/gpu/drm/i915/intel_sdvo.c
index fa8c785..cfcc2a8 100644
--- a/drivers/gpu/drm/i915/intel_sdvo.c
+++ b/drivers/gpu/drm/i915/intel_sdvo.c
@@ -1480,7 +1480,7 @@ intel_sdvo_hdmi_sink_detect(struct drm_connector *connector)
 	if (edid == NULL)
 		edid = intel_sdvo_get_analog_edid(connector);
 
-	status = connector_status_disconnected;
+	status = connector_status_unknown;
 	if (edid != NULL) {
 		/* DDC bus is shared, match EDID to connector type */
 		if (edid->input & DRM_EDID_INPUT_DIGITAL) {
-- 
1.7.6

